python/paddle/fluid/dygraph/dygraph_to_static/logging_utils.py

```diff
@@ -0,0 +1,225 @@
+
+VERBOSITY_ENV_NAME = 'TRANSLATOR_VERBOSITY'
+DEFAULT_VERBOSITY = 0
+
+
+def synchronized(func):
+    def wrapper(*args, **kwargs):
+        with threading.Lock():
+            return func(*args, **kwargs)
+
+    return wrapper
+
+
+class TranslatorLogger(object):
+    """
+    class for Logging and debugging during the tranformation from dygraph to static graph.
+    The object of this class is a singleton.
+    """
+
+    @synchronized
+    def __new__(cls, *args, **kwargs):
+        if not hasattr(cls, '_instance'):
+            cls._instance = object.__new__(cls, *args, **kwargs)
+            cls._instance._initialized = False
+        return cls._instance
+
+    def __init__(self):
+        if self._initialized:
+            return
+
+        self._initialized = True
+        self._logger = log_helper.get_logger(
+            __name__, 1, fmt='%(asctime)s-%(levelname)s: %(message)s')
+        self._verbosity_level = None
+        self._transformed_code_level = None
+
+    @property
+    def logger(self):
+        return self._logger
+
+    @property
+    def verbosity_level(self):
+        if self._verbosity_level is not None:
+            return self._verbosity_level
+        else:
+            return int(os.getenv(VERBOSITY_ENV_NAME, DEFAULT_VERBOSITY))
+
+    @verbosity_level.setter
+    def verbosity_level(self, level):
+        self.check_level(level, {})
+        self._verbosity_level = level
+
+    @property
+    def transformed_code_level(self):
+        if self._transformed_code_level is not None:
+            return self._transformed_code_level
+        else:
+            return -1
+
+    @transformed_code_level.setter
+    def transformed_code_level(self, level):
+        self.check_level(level, _transformer_name_to_level)
+        self._transformed_code_level = level
+
+    def check_level(self, level, name_to_level_dict):
+        if isinstance(level, (six.integer_types, type(None))):
+            rv = level
+        elif str(level) == level:
+            if level not in name_to_level_dict:
+                raise ValueError("Unknown level: %r" % level)
+            rv = name_to_level_dict[level]
+        else:
+            raise TypeError("Level not an integer or a valid string: %r" %
+                            level)
+        return rv
+
+    def has_code_level(self, level):
+        level = self.check_level(level, _transformer_name_to_level)
+        return level == self.transformed_code_level
+
+    def has_verbosity(self, level):
+        level = self.check_level(level, {})
+        return level >= self.verbosity_level
+
+    def error(self, msg, *args, **kwargs):
+        self.logger.error(msg, *args, **kwargs)
+
+    def warn(self, msg, *args, **kwargs):
+        self.logger.warn(msg, *args, **kwargs)
+
+    def log(self, level, msg, *args, **kwargs):
+        if self.has_verbosity(level):
+            self.logger.log(level, msg, *args, **kwargs)
+
+    def log_transformed_code(self, level, ast_node, *args, **kwargs):
```

Comment:
1. You level is not an integer but a string, level at other places is integer, you had better to rename it "transformer_name" or something similar to it.

2. As I commented, could we delete "_transformer_name_to_level", just keep an integer and you increment the integer by 1 every time you call this method? Then your "has_code_level" can compare the integer and the code level set by user.

3. msg can add "transformer_name". For example:

msg = "After the {} -st/rd/th {}, transformed code:\n{}".format(the_integer, transformer_name, ast_to_source_code(ast_node))

Then users or Paddle develop can know the mapping of their level to transformer_name, and transformer code

Reply:
Thanks, done